packages/atlas/src/views/global/NftSaleBottomDrawer/NftForm/SetUp/SetUp.tsx

```diff
@@ -154,66 +163,66 @@ export const SetUp: React.FC<SetUpProps> = ({
+                    />
+                  )}
                 />
-                <AuctionDatePicker
-                  size="regular"
-                  label="expiration date"
-                  minDate={new Date()}
-                  disabled={!activeInputs.includes('auctionDuration') || !auctionDate.startDate}
-                  onChange={(value) => setAuctionDuration({ endDate: value })}
-                  items={[
-                    {
-                      value: auctionDate.startDate && addDays(new Date(auctionDate.startDate), 1),
-                      name: '1 day',
-                    },
-                    {
-                      value: auctionDate.startDate && addDays(new Date(auctionDate.startDate), 3),
-                      name: '3 days',
-                    },
-                    {
-                      value: auctionDate.startDate && addDays(new Date(auctionDate.startDate), 5),
-                      name: '5 days',
-                    },
-                    {
-                      value: auctionDate.startDate && addDays(new Date(auctionDate.startDate), 7),
-                      name: '7 days',
-                    },
-                  ]}
-                  value={auctionDate.endDate}
+                <Controller
+                  name="endDate"
+                  control={control}
+                  render={({ field: { onChange, value }, fieldState: { error } }) => (
+                    <AuctionDatePicker
+                      size="regular"
+                      label="expiration date"
+                      error={!!error}
+                      helperText={error?.message}
+                      minDate={(startDate?.type === 'date' && startDate.date) || new Date()}
+                      disabled={!activeInputs.includes('auctionDuration')}
+                      onChange={onChange}
+                      items={expirationDateItems}
+                      value={value || { durationDays: null, type: 'duration' }}
+                    />
+                  )}
                 />
               </AuctionDatePickerWrapper>
             </FormField>
-            {auctionDate.startDate instanceof Date && auctionDate.endDate instanceof Date && (
+            {numberOfBlocksAndDaysLeft && numberOfBlocksAndDaysLeft.blocks > 0 && (
               <DaysSummary>
                 <Text variant="t200-strong" color={cVar('colorTextMuted', true)}>
                   Total:
                 </Text>
                 &nbsp;
-                <Text variant="t200-strong">{daysAndHours}</Text>
+                <Text variant="t200-strong">{numberOfBlocksAndDaysLeft.daysAndHoursText}</Text>
                 &nbsp;
                 <Text variant="t200-strong" secondary>
-                  / {numberOfBlocks?.toLocaleString('no', { maximumFractionDigits: 1 })} Blocks
+                  / {numberOfBlocksAndDaysLeft.blocks?.toLocaleString('no', { maximumFractionDigits: 1 })} Blocks
```

Comment:
Let's update the implementation of `formatNumber` in `@/utils/number` and use it here